oe_contact_forms.module

```diff
@@ -6,3 +6,327 @@
+use Drupal\contact\ContactFormInterface;
+
+/**
+ * Implements hook_form_FORM_ID_alter() for contact_form_form().
+ */
+function oe_contact_forms_form_contact_form_form_alter(&$form, FormStateInterface $form_state) {
+  /** @var \Drupal\contact\ContactFormEditForm $form_object */
+  $form_object = $form_state->getFormObject();
+
+  if (!in_array($form_object->getOperation(), ['edit', 'add'], TRUE)) {
+    // Only alter the edit and add forms.
+    return;
+  }
+
+  /** @var \Drupal\contact\ContactFormInterface $contact_form */
+  $contact_form = $form_object->getEntity();
+
+  $form['is_corporate_form'] = [
+    '#type' => 'checkbox',
+    '#title' => t('Is corporate form'),
+    '#description' => t("Check this box if you'd like to make this a corporate form."),
+    '#default_value' => $contact_form->getThirdPartySetting('oe_contact_forms', 'is_corporate_form', FALSE),
+  ];
+
+  $form['#tree'] = TRUE;
+  $form['topics_fieldset'] = [
+    '#type' => 'fieldset',
+    '#title' => t('Topics'),
+    '#states' => [
+      'visible' => [
+        ':input[name="is_corporate_form"]' => ['checked' => TRUE],
+      ],
+    ],
+    '#prefix' => '<div id="topics-fieldset-wrapper">',
+    '#suffix' => '</div>',
+    '#description' => t("Please select the topics and the associated email addresses the users can choose when submitting the form. For multiple email addresses, please separate them with a comma."),
+  ];
+
+  $topic_names = $contact_form->getThirdPartySetting('oe_contact_forms', 'topic_name', []);
+  $topic_email_addresses = $contact_form->getThirdPartySetting('oe_contact_forms', 'topic_email_address', []);
+  $num_topics = $form_state->get('num_topics');
+  $n = count($topic_names);
+
+  // No values added yet.
+  if ($num_topics === NULL && $n === 0) {
+    $num_topics = 1;
+  }
+
+  // Any saved values.
+  if ($num_topics === NULL && $n > 0) {
+    $num_topics = $n;
+  }
+
+  $form_state->set('num_topics', $num_topics);
+
+  for ($i = 0; $i < $num_topics; $i++) {
+    $form['topics_fieldset']['group'][$i]['topic_name'] = [
+      '#type' => 'textfield',
+      '#title' => t('Topic name'),
+      '#states' => [
+        'required' => [
+          ':input[name="is_corporate_form"]' => ['checked' => TRUE],
+        ],
+      ],
+      '#default_value' => isset($topic_names[$i]) ? $topic_names[$i] : '',
+    ];
+    $form['topics_fieldset']['group'][$i]['topic_email_address'] = [
+      '#type' => 'textfield',
+      '#title' => t('Topic email address(es)'),
+      '#states' => [
+        'required' => [
+          ':input[name="is_corporate_form"]' => ['checked' => TRUE],
+        ],
+      ],
+      '#default_value' => isset($topic_email_addresses[$i]) ? $topic_email_addresses[$i] : '',
+    ];
+  }
+
+  $form['topics_fieldset']['actions'] = [
+    '#type' => 'actions',
+  ];
+  $form['topics_fieldset']['actions']['add_topic'] = [
+    '#type' => 'submit',
+    '#value' => t('Add topic'),
+    '#submit' => ['_oe_contact_forms_add_one_callback'],
```

Comment:
You have to limit the validation errors here because it validates the entire form when this is submitted. Also when removing one. For example:

```
'#limit_validation_errors' => [],
```

Reply:
It was changed by the ajax refactoring but actually this key is added.